### PR TITLE
Updating Dashboard guildelines document

### PR DESCRIPTION
**Pull request by gizas**
## What does this PR do?
Update Dashboard Guidelines Documentation 

Related Docs for reference:
- https://docs.google.com/document/d/1uyyFGx6xA5Kvl8c-ZdvXdvBGrHTylxU9F69TGqfzdmw/edit#
- https://docs.google.com/document/d/1_7ZYOO5KsSG4J3ovdlB4obBF5i0FQfsiXrRtI3wWFpg/edit#

### CODE REVIEW

File: docs/dashboard_guidelines.md [hunk 1]
```diff
@@ -9,34 +9,123 @@ The goal of each integration dashboard is to:
...
+### Add Visualizations by value, not by reference inside a dashboard
 
 Kibana visualizations can be added in a dashboard by value or by reference. Historically by value did not exist. Switching to value has the advantage that the dashboards are fully self contained and only need a single request to be installed.
 
 To achieve this:
-- Migrate existing dashboards from by reference to by value.
-- Create new dashboards adding visualizations by value.
+
+* Migrate existing dashboards from by reference to by value
+* Create new dashboards adding visualizations by value
```
Comment (ruflin) [inline docs/dashboard_guidelines.md:36]: Do we have a docs link here?

**Review by ruflin — APPROVED** [review]
LGTM. lets get this in an keep iterating on it.

It seems this fully replaces one of the google docs, parts of the other? Lets make sure we add a note to the google docs that the content is now living here.

File: docs/dashboard_guidelines.md [hunk 2]
```diff
@@ -9,34 +9,123 @@ The goal of each integration dashboard is to:
...
+
+Migrate the dashboards from TSVB to Lens where possible. If it's not possible, please engage with the Kibana team to identify any gaps that prevent full TSVB to Lens dashboard migration.
+
+### Visualizations should contain a filter
 
 Kibana visualizations can define a filter to avoid performance issues querying all `metrics-*` or `logs-*` indices.
 
 It is recommended to set a filter in each visualization at least by the required `data_stream.dataset`. More details about the Elastic data stream naming scheme [here][3].
 
+Avoid using general filters as possbile (filters with `-*`).  Combine multpiple fileds and values inside a filter with AND/OR operators. Altough your filter might become more complex, will avoid extra queries.
```
Comment (mrodm) [inline docs/dashboard_guidelines.md:90]:
```suggestion
Avoid using general filters as possible (filters with `-*`).  Combine multiple fields and values inside a filter with AND/OR operators. Although your filter might become more complex, will avoid extra queries.
```

File: docs/dashboard_guidelines.md [hunk 3]
```diff
@@ -9,34 +9,123 @@ The goal of each integration dashboard is to:
...
 
 ![filter in visualization](./images/filter_in_visualization.png)
 
+### Do not use “library visualizations”
+
+Do not use the visualisations that show up in the `Analytics > Visualize library`. Instead define visualizations as part of the dashboard. This is the default when creating new panels by clicking “Add new visualization” on the dashboard. If some panels are already saved to the library, you can unlink them and delete them from the library. There are little use cases where library visualizations are preferable. It makes sense e.g. if a given visualization always has to be exactly the same on multiple dashboards or if its users frequently look at the visualization without looking at the whole dashboard.
+
+### Use dashboard-native controls
+
+The **Input controls** visualization type is deprecated in favor of **Controls** embedded into the dashboard itself. The `Controls` dropdown in the Dashboard menu bar should be used. See the [documentation](https://www.elastic.co/guide/en/kibana/master/add-controls.html) page for more information
```
Comment (mrodm) [inline docs/dashboard_guidelines.md:102]:
```suggestion
The **Input controls** visualization type is deprecated in favor of **Controls** embedded into the dashboard itself. The `Controls` dropdown in the Dashboard menu bar should be used. See the [documentation](https://www.elastic.co/guide/en/kibana/master/add-controls.html) page for more information.
```

**Review by mrodm — APPROVED** [review]
Thanks for adding these guidelines! :+1: 
Just added a couple of suggestions.

File: docs/dashboard_guidelines.md (hunk 1 above)
Comment (MichaelKatsoulis) [reply docs/dashboard_guidelines.md:36]: Should we mention this script ? https://github.com/flash1293/legacy_vis_analyzer

**Review by MichaelKatsoulis — APPROVED** [review]

File: docs/dashboard_guidelines.md (hunk 1 above)
Comment (gizas) [reply docs/dashboard_guidelines.md:36]: > sure we add a note to the google docs that the content is now living here.

Updated docs

Comment (mrodm) [reply docs/dashboard_guidelines.md:36]: > @mrodm might have any more info on the above?

I don't know of any doc that could be added here that suggests to use dashboards by reference, sorry

Comment (ruflin) [reply docs/dashboard_guidelines.md:36]: The doc link I was referring to was more about docs how to create by value not about "why by value". Why we switch to by value is documented in some github issue but I don't think we have to reference these here.

**Comment by mrodm** [comment]
@gizas CI failure is unrelated to the changes of this PR. They have been reported too https://github.com/elastic/integrations/issues/5222

**Pull request merged**